src/Symfony/Component/DependencyInjection/Tests/Compiler/AutowirePassTest.php

```diff
@@ -405,6 +406,9 @@ public function testResolveParameter()
         $this->assertEquals(Foo::class, $container->getDefinition('bar')->getArgument(0));
     }
 
+    /**
+     * @group legacy
```

Comment:
Why was this change necessary?

Reply:
This test case is specifically about the deprecated syntax, testing autowiring of arguments with default value positioned before a required one.

Reply:
To get rid of a wanted deprecation, see https://github.com/symfony/symfony/pull/54308#discussion_r1528733421 for the details 🙂

Reply:
Understood, but why wasn't the legacy group required before?

Reply:
I don't know. It is deprecated since 8.1 (https://3v4l.org/n4leq), however I wouldn't be able to explain why this warning pops now 🤔

Reply:
This was not deprecated in 8.1 but has just been in 8.4. It was not deprecated because that's the only syntax that was once supported on older versions to make a type nullable.

Reply:
https://3v4l.org/K8qWH

Reply:
Thanks for the info 👍